lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -6,32 +6,31 @@
 library js_promise;
 
 import 'dart:js_interop';
-
-import 'package:js/js_util.dart' as js_util;
-
-import '../util.dart';
-
-extension CallExtension on JSFunction {
-  external void call(JSAny? this_, JSAny? object);
-}
-
-@JS('Promise')
-external JSAny get _promiseConstructor;
-
-JSPromise<JSAny?> createPromise(JSFunction executor) =>
-  js_util.callConstructor(
-    _promiseConstructor,
-    <Object>[executor],
-  );
-
-
-JSPromise<JSAny?> futureToPromise<T extends JSAny?>(Future<T> future) {
-  return createPromise((JSFunction resolver, JSFunction rejecter) {
-    future.then(
-      (T value) => resolver.call(null, value),
-      onError: (Object? error) {
-        printWarning('Rejecting promise with error: $error');
-        rejecter.call(null, null);
+import 'dart:js_interop_unsafe';
+
+/// This is the same as package:js_interop's FutureToPromise (.toJS), but with
+/// a more descriptive error message.
+///
+/// TODO(srujzs/dit): Move to js_interop's .toJS.
+extension CustomFutureOfJSAnyToJSPromise<T extends JSAny?> on Future<T> {
+  /// A [JSPromise] that either resolves with the result of the completed
+  /// [Future] or rejects with an object that contains its error.
+  JSPromise<T> get toPromise {
+    return JSPromise<T>((JSFunction resolve, JSFunction reject) {
```

Comment:
Hmm, the only functional difference between this and `toJS` I see is that the `then` value here doesn't return anything in either the `onValue` or `onError` functions, but it would be strange if that makes a difference. Are you still seeing the same issues with `toJS`? Would the sane issue manifest with `toPromise` if we returned `value` and `wrapper` in the `onValue` and `onError` functions here?